src/main/resources/templates/adminDash.html

```diff
@@ -41,21 +41,31 @@ <h3>Customers</h3>
               <th> End Date </th>
           </tr>
 
-          <tr>
-              <td th:text="${username}" id="username">username</td>
-              <td th:text="${name}" id="name">name</td>
-              <td th:text="${password}" id="password">pw</td>
-              <td th:text="${subscription}" id="subscription">
+          <tr th:each="customer : ${customers}">
+              <td th:text="${customer.username}"></td>
+              <td th:text="${customer.name}"></td>
+              <td th:text="${customer.password}"></td>
+              <td>
 
                   <div class="select_dropdown">
                       <select>
-                          <option th:text="${subscription}" id="option1"> Free </option>
-                          <option th:text="${subscription}" id="option2"> Paid </option>
+                          <option th:text="${customer.subscription}" id="option1"> </option>
+                          <option id="option2"></option>
+                          <script>
+                              if (document.getElementById("option1").innerHTML === "TRIAL") {
+                                  document.getElementById("option2").innerHTML = "PAID";
+                              }
+                              if (document.getElementById("option1").innerHTML=== "PAID") {
+                                  document.getElementById("option2").innerHTML = "TRIAL";
+                              }
```

Comment:
This is in a loop, doesn't that mean you will have multiple options with `id=option2`? IDs are meant to be unique, it might not cause a crash, but the script surely will modify all the options at once which doesn't seem to be what you intended.

Testing with two users:
![image](https://user-images.githubusercontent.com/1709432/161895292-094b76c5-baae-405a-addd-417fbe5283a0.png)

Also I'm assuming these don't actually do anything. I think you're doing it the hard way, there's no need for the end date to by modifiable according to the problem description, and you can use a button for the upgrade/downgrade instead of these selection boxes.

If you modify the upgrade and downgrade endpoints to take the username, you could use this template:

```html
<a th:if="${user.getSubscription()}" th:href="'/downgrade/' + user.getUsername()"><button>Downgrade</button></a>
<a th:if="${not user.getSubscription()}" th:href="'/upgrade/' + user.getUsername()"><button>Upgrade</button></a>
```